### PR TITLE
Use declarative pipeline

### PR DESCRIPTION
**Pull request by v1v**
### What

Use declarative pipeline, for such I had to add a couple of wrappers to simplify the usage of `script` closures. 

Fail fast for linux since it does not create parallel stages for each linux flavour. There is a limitation for parallel in conjunction with matrix :(

### Actions

- [x] Skip cron for PRs but only the master branch

#### Untrusted CI

image

#### Trusted CI

image

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,134 +1,153 @@
...
+                    }
+                    stages {
+                        stage('Build') {
+                            steps {
+                                sh 'make build'
+                            }
+                        }
+                        stage('Test') {
+                            steps {
+                                sh 'make test-debian test-slim test-alpine test-jdk11 test-centos test-centos7'
```
Comment (slide) [inline Jenkinsfile:89]: When we move to multi-arch, this becomes much more complex because we will have different architectures as well.

Comment (v1v) [reply Jenkinsfile:89]: Do you think the matrix might help? I initially thought to use it, but the limitation of parallel with matrix wasn't a kind of blocker, bu knowing the upcoming work, so I'm happy to give another go to use just a matrix to support some other cases or if you prefer i can jump in to the next syncup meeting :)

Comment (v1v) [reply Jenkinsfile:89]: I've just found https://github.com/jenkinsci/docker/pull/998

So if I will enable it here

Comment (v1v) [reply Jenkinsfile]: I just committed https://github.com/jenkinsci/docker/pull/1023/commits/d7d41b8e682dfab0582346cc6aa9cf03295d84dd to support a matrix configuration with a bunch of exclusions based on the information I gathered from #998

**Comment by dduportal** [comment]
Hello @v1v , what is the status of this PR? Is there anything we could do to help you to land this PR?

**Comment by timja** [comment]
> Hello @v1v , what is the status of this PR? Is there anything we could do to help you to land this PR?

looks like it just needs conflicts resolved and then someone with write access modifying the file / re-submitting to make sure it is working fine

**Review by DavidRomanovizc — APPROVED** [review]
LGTM!

**Review by DavidRomanovizc — APPROVED** [review]

**Comment by v1v** [comment]
My notifications are silent and I missed this PR and feedback, sorry about this. I'll solve the conflicts in the next few days.

Thanks

**Comment by v1v** [comment]
As far as I see there is no such attraction that makes me thing I should spend further time resolving the conflicts again, so I'll close this PR.

if someone would like to contribute feel free to reopen it or cherry pick the changes.

**Pull request closed without merging**